src/ThreeEditor/util/Materials/MaterialsManager.ts

```diff
@@ -14,56 +13,76 @@ export interface MaterialsManagerJSON {
     opacity: number,
     transparent: boolean
 }
-export default class MaterialsManager{ 
-    private prefabMaterials: Record<string,SimulationMaterial>;
-    private customMaterials: Record<string,SimulationMaterial>;
-    materialOptions: Record<string,string>;
+export default class MaterialsManager {
+    private prefabMaterials: Record<string, SimulationMaterial>;
+    private customMaterials: Record<string, SimulationMaterial>;
+    materialOptions: Record<string, string>;
     materials: Record<string, SimulationMaterial>;
-    constructor(){
+
+    constructor() {
         const [materials, options] = this.createMaterialPrefabs();
         this.prefabMaterials = materials;
         this.materialOptions = options;
         this.customMaterials = {};
-        const materialsHandler = {
-            get: (target:Record<string,SimulationMaterial>, prop:string) => {
 
-                return (prop && prop in target
-                    ? target[ prop ]
-                    : (prop && prop in this.prefabMaterials
-                    ? this.prefabMaterials[ prop ]
-                    : this.getDefaultMaterial()));
+        const materialsHandler = {
+            get: (target: Record<string, SimulationMaterial>, prop: string) => {
+                return (prop in target
+                    ? target[prop]
+                    : (prop in this.prefabMaterials
+                        ? this.prefabMaterials[prop]
+                        : this.getDefaultMaterial()));
             },
-            set: (target:Record<string,SimulationMaterial>, prop:string, value:unknown) => {
+            set: (target: Record<string, SimulationMaterial>, prop: string, value: unknown) => {
 
                 Reflect.set(this.materialOptions, prop, prop);
                 return Reflect.set(target, prop, value);
             },
+            ownKeys: (target: Record<string, SimulationMaterial>) => {
+                
+                return Array.from(new Set([...Reflect.ownKeys(target), ...Reflect.ownKeys(this.prefabMaterials)]));
+            },
+            has: (target: Record<string, SimulationMaterial>, key: string) => {
+                return key in target || key in this.prefabMaterials;
```

Comment:
```suggestion
                return key in this.prefabMaterials || key in target;
```
prefabs are more likely to have searched key